packages/eas-cli/src/build/build.ts

```diff
@@ -173,26 +173,23 @@ function handleBuildRequestError(error: any, platform: Platform): never {
   Log.debug(JSON.stringify(error.graphQLErrors, null, 2));
 
   if (SERVER_SIDE_DEFINED_ERRORS.includes(error?.graphQLErrors?.[0]?.extensions?.errorCode)) {
-    Log.error(error?.graphQLErrors?.[0]?.message);
-    throw new Error('Build request failed.');
+    throw new Error(error?.graphQLErrors?.[0]?.message);
   } else if (
     error?.graphQLErrors?.[0]?.extensions?.errorCode === 'EAS_BUILD_DOWN_FOR_MAINTENANCE'
   ) {
-    Log.error(
+    throw new Error(
```

Comment:
Consider creating custom error class, so you can latter distinguish error with proper use facing error from all the others

Reply:
@wkozyra95 Created some custom error classes for previously defined errors in case a custom behaviour would be needed in the future

Reply:
Sorry should have been more clear. I meant to suggest creating sth like `EasCommandError` for errors that we now are already properly formated and ready to print as it is. So in the code that it prints you could do

```
if (err instanceof EasCommandError) {
  //print
} else {
  // rest of the logic for printing unknown erorrs
}
```

I don't have anything against making it more granular, so I'm fine with that change, but with this implementation there still no way to detect if error needs to be processed or not.